pkg/trace/agent/agent.go

```diff
@@ -206,6 +206,10 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 			// Figure out the top-level spans and sublayers now as it involves modifying the Metrics map
 			// which is not thread-safe while samplers and Concentrator might modify it too.
 			traceutil.ComputeTopLevel(t)
+		} else {
+			for _, span := range t {
```

Comment:
Yet another trace loop? Perhaps you can find another loop to squeeze this in.

Reply:
moved on another trace loop above